runtime/turing/src/migrations.rs

```diff
@@ -0,0 +1,526 @@
+			let last_asset_id: TokenId = (assets.len() - 1).try_into().unwrap();
+			LastAssetId::<Runtime>::set(Some(last_asset_id));
+
+			log::info!(
+				target: "asset_registry",
+				"on_runtime_upgrade: New data inserted"
+			);
+
+			// Each asset + each asset location + updating last asset id
+			let total_rw = assets.len() as u32 * 2 + 1;
+			<Runtime as frame_system::Config>::DbWeight::get()
+				.reads_writes(total_rw as Weight, total_rw as Weight)
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			log::info!(
+				target: "asset_registry",
+				"pre_upgrade check"
+			);
+
+			// Assert asset Metadata length is 0
+			let asset_metadata_count =
+				orml_asset_registry::Metadata::<Runtime>::iter().collect::<Vec<_>>().len();
+			assert_eq!(asset_metadata_count, 0);
+
+			// Assert asset LocationToAssetId length is 0
+			let location_to_asset_id_count =
+				orml_asset_registry::LocationToAssetId::<Runtime>::iter()
+					.collect::<Vec<_>>()
+					.len();
+			assert_eq!(location_to_asset_id_count, 0);
+
+			// Assert last asset id is 0
+			if let Some(_last_asset_id) = LastAssetId::<Runtime>::get() {
+				return Err("pre_upgrade check - LastAssetId should equal None".into())
+			}
+
+			Ok(())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			log::info!(
+				target: "asset_registry",
+				"post_upgrade check"
+			);
+
+			// Assert Metadata length
+			let asset_metadata_count =
+				orml_asset_registry::Metadata::<Runtime>::iter().collect::<Vec<_>>().len();
+			assert_eq!(asset_metadata_count, 9);
+
+			// Assert LocationToAssetId length
+			let location_to_asset_id_count =
+				orml_asset_registry::LocationToAssetId::<Runtime>::iter()
+					.collect::<Vec<_>>()
+					.len();
+			assert_eq!(location_to_asset_id_count, 9);
+
+			// Assert last asset id (zero index)
+			if let Some(last_asset_id) = LastAssetId::<Runtime>::get() {
+				assert_eq!(last_asset_id, 8);
+			} else {
+				return Err("post_upgrade check - LastAssetId should equal 8".into())
+			}
+
+			Ok(())
+		}
+	}
+
+	pub struct TokensCurrencyIdMigration;
+	impl OnRuntimeUpgrade for TokensCurrencyIdMigration {
+		fn on_runtime_upgrade() -> Weight {
+			log::info!(
+				target: "orml_tokens",
+				"on_runtime_upgrade: Attempted to apply tokens"
+			);
+
+			// Migrate Tokens Accounts CurrencyId from Enum to u32
+			let mut tokens_accounts: Vec<(AccountId, TokenId, orml_tokens::AccountData<Balance>)> =
+				vec![];
+			Accounts::drain().for_each(|(account_id, currency_id, account_data)| {
+				tokens_accounts.push((account_id, currency_id as u32, account_data));
+			});
+			tokens_accounts.iter().for_each(|(account_id, currency_id, account_data)| {
+				orml_tokens::Accounts::<Runtime>::insert(account_id, currency_id, account_data);
+			});
+			log::info!(
+				target: "orml_tokens",
+				"on_runtime_upgrade: Accounts data updated {:?}",
+				orml_tokens::Accounts::<Runtime>::iter().count(),
```

Comment:
Does this cause a full set of extra reads? I'm not sure how substrate does `iter().count()`.